cloudant-sync-datastore-core/src/main/java/com/cloudant/common/RetriableTask.java

```diff
@@ -62,26 +60,27 @@ public int getTriesRemaining() {
 
     @Override
     public T call() throws Exception {
-        while (true) {
+        Exception lastException = null; // remember the exception we caught from the last try
+        do {
             try {
-                logger.fine("Retry #: " + triesRemaining + "," + task.toString());
+                logger.fine(String.format("%d tries remaining for task %s", triesRemaining, task));
                 T t = task.call();
-                if(triesRemaining < 3) {
-                    logger.fine("Success after retry: " + triesRemaining + ", " + task.toString());
-                }
+                logger.fine(String.format("Success with %d tries remaining for task %s",
+                        triesRemaining, task));
```

Comment:
I think there is an off by one issue here because the successful attempt has consumed one of the tries. In which case the log message should be `triesRemaining - 1`. Although arguably since the field and parameter are called `totalRetries` not `totalTries` then the message is actually correct and the `while` condition should be `--triesRemaining >= 0`.